src/mainwindow.cpp

```diff
@@ -398,3 +488,80 @@ void MainWindow::aboutHotspot()
     dialog.adjustSize();
     dialog.exec();
 }
+
+void MainWindow::navigateToCode(const QUrl &url, int lineNumber, int columnNumber)
```

Comment:
I don't see us supporting remote urls, so please change the url to a `const QString &path`